src/components/Email.tsx

```diff
@@ -24,6 +25,10 @@ const Email = () => {
 
         if (invalidationRules.some(rule => rule.test(value))) {
             setError('Please enter a valid email address');
+        } else if (await isEmailExists(value)) {
```

Comment:
please, not use `await` inside if-clause. Calculate it before, and use result into the if

```
const isExists = isEmailExists(value);
if (isExists) {...}
```

reason - will be easier to add `console.log` or `debugger` to check result of `isEmailExists`

Reply:
done